src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/Config.java

```diff
@@ -627,6 +638,17 @@ private int parseIntegerEntry(String keyword) throws IOException {
         return value;
     }
 
+    private <E extends Enum<E>> E parseEnumEntry(Class<E> enumClass,
+            String keyword) throws IOException {
+        String value = parseStringEntry(keyword);
+        try {
+            return Enum.valueOf(enumClass, value);
+        } catch (IllegalArgumentException ignored) {
+            throw excToken(keyword + " must be one of " +
```

Comment:
nit: since we are using the `value` for enum conversion, maybe using it instead of `keyword` for the error message? This has the benefit of showing what the parsed value is.

Reply:
The `value` is completed form the current token by `excToken()`.

As mentioned in https://github.com/openjdk/jdk/pull/18898#issuecomment-2138360201, if we pass `CS4` (an invalid value), we get the following error:
```
cipherTextStealingVariant must be one of [CS1, CS2, CS3], read: Token[CS4], line 33
```

Other attributes are parsed in the same way, with similar error messages.

Reply:
Ok, sounds good.